exchanges/gateio/gateio_wrapper.go

```diff
@@ -205,31 +195,101 @@ func (g *Gateio) Setup(exch *config.Exchange) error {
 		return err
 	}
 
-	wsRunningURL, err := g.API.Endpoints.GetURL(exchange.WebsocketSpot)
+	err = g.Websocket.Setup(&stream.WebsocketSetup{
+		ExchangeConfig:               exch,
+		Features:                     &g.Features.Supports.WebsocketCapabilities,
+		FillsFeed:                    g.Features.Enabled.FillsFeed,
+		TradeFeed:                    g.Features.Enabled.TradeFeed,
+		UseMultiConnectionManagement: true,
+	})
+	if err != nil {
+		return err
+	}
+	// Spot connection
+	err = g.Websocket.SetupNewConnection(&stream.ConnectionSetup{
```

Comment:
If we do that then if we enable, disable or re-enable an asset then we might need to call setup again. I can't think of a better way of doing it. Let me know your thoughts on how to over come that little obstacle because what I have done is not the best.

Reply:
I don't think even in its current form, the scenario you speak of doesn't work. If it is important to you, then you already need to change code. t present.
Enabling an asset does not call `exch.Websocket.Connect`, the websocketroutinemanager already considers the websocket connected, it only checks for ws.enabled. I don't find it as important to consider, especially since it doesn't work.

In fact, you raising this has demonstrated a bug!!!!!!
```
gctcli % go run . disableexchange gateio 
{
 "status": "success"
}%                        
```

```
[DEBUG] | WEBSOCKET | 12/09/2024 16:16:42 | GateIO websocket: running connection monitor cycle
[DEBUG] | WEBSOCKET | 12/09/2024 16:16:42 | GateIO websocket: shutting down websocket
[DEBUG] | WEBSOCKET | 12/09/2024 16:16:42 | GateIO websocket: completed websocket shutdown
[INFO]  | EXCHANGE | 12/09/2024 16:16:42 | gateio exchange unloaded successfully.
[INFO]  | WEBSOCKET | 12/09/2024 16:16:55 | GateIO Websocket connected to wss://api.gateio.ws/ws/v4/
[DEBUG] | WEBSOCKET | 12/09/2024 16:16:55 | GateIO websocket connection: sending message [{"time":1726121815,"id":5,"channel":"spot.tickers","event":"subscribe","payload":["BTC_USDT"]}]
...
```

Reply:
As for the asset enabled aspect. I think I would just like to see something more explicit than relying on `GenerateSubs()`returning asset error to prevent a ws from connecting.
I don't mind it if Gateio setup stays the same ultimately

Reply:
I removed the asset error handling in recent commits and defaulted to what spot was doing.

Reply:
I fixed the flushing in the most recent commit and should be good to go, I can't consolidate the code down to what it was before as it is too different from previous handling to multiconnection handling so I reverted that and expanded test functionality in `TestFlushChannels`; [check here](https://github.com/thrasher-corp/gocryptotrader/pull/1580/commits/0709ba30eaab1b702eb19bc0230dc4100ac55215)

Reply:
As for the unloading and loaded the exchange via RPC there seems to be a bit more in engine that needs to be done to support this; might have to open another PR to fix this issue.